### PR TITLE
Document Random Beacon v2 parameters

### PR DESCRIPTION
**Pull request by nkuba**
This PR documents Random Beacon V2 parameters.

It is a follow-up of https://github.com/keep-network/keep-core/pull/2746#discussion_r760781785.

### CODE REVIEW

**Comment by pdyraga** [comment]
Heads up, initial values depends on https://github.com/keep-network/keep-core/pull/2749.

File: solidity/random-beacon/README.adoc [hunk 1]
```diff
@@ -289,6 +286,144 @@ operators whose nodes never participate in signing because they are
...
+|
+
+|dkgMaliciousResultNotificationRewardMultiplier
+|Percentage of the staking contract malicious behavior notification reward which
+will be transferred to the notifier reporting about a malicious DKG result
+|Yes
+|
+
+|===
+
```
Comment (tomaszslabon) [inline solidity/random-beacon/README.adoc:426]: We also have `minimumAuthorization` and `authorizationDecreaseDelay` parameters, but they don't seem to be used anywhere.
CC @pdyraga

Comment (pdyraga) [reply solidity/random-beacon/README.adoc:426]: They will be used for T staking contract integration. This work is still in progress.

Comment (tomaszslabon) [reply solidity/random-beacon/README.adoc:426]: Should they be included in this list of parameters?

**Comment by tomaszslabon** [comment]
I don't have any comments except the possible additions of `minimumAuthorization` and `authorizationDecreaseDelay` parameters and updating the empty values from Piotr's PR.

File: solidity/random-beacon/README.adoc (hunk 1 above)
Comment (pdyraga) [reply solidity/random-beacon/README.adoc:426]: Let's add them later, once we are sure they do not need to be updated in #2747.

File: solidity/random-beacon/README.adoc [hunk 2]
```diff
@@ -289,6 +286,144 @@ operators whose nodes never participate in signing because they are
...
+|Yes
+|
+
+|ineligibleOperatorNotifierReward
+|Reward in T for notifying the operator is ineligible
+|Yes
+|
+
+|maliciousDkgResultSlashingAmount
+|Slashing amount for supporting malicious DKG result
```
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:392]: Are we slashing all group members who signed (supported) the malicious result or are we slashing only the submitter of the malicious result? I thought that we're slashing for supporting, but according to line 70 we're slashing for submitting.
```suggestion
|Slashing amount for submitting malicious DKG result
```

File: solidity/random-beacon/README.adoc [hunk 3]
```diff
@@ -197,11 +196,10 @@ rewards.
 [[fees-and-rewards]]
 === Fees and Rewards
 
-Relay requester should provide a fee in T. The value of the fee is a governable
-parameter. The entire fee is deposited in the DKG rewards pool that is used to
-reimburse for different actions related to DKG.
+Relay requester should provide a fee in T. The entire fee is deposited in the DKG
+rewards pool that is used to reimburse for different actions related to DKG.
```
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:200]: The fee for submitting a relay request will be governable, right? Isn't it missing it from the list?

File: solidity/random-beacon/README.adoc [hunk 4]
```diff
@@ -68,15 +68,14 @@ malicious calling `RandomBeacon.challengeDkgResult(DKG.Result calldata dkgResult
...
-and slashing amount are governable parameters.
+malicious result is immediately discarded.
 
 Once the challenge period passes, and no challenges are reported,
 the DKG result submitter should unlock the sortition pool and mark the DKG result as
 accepted calling
 `RandomBeacon.approveDkgResult(DKG.Result calldata dkgResult)` to receive a
 reward. In case the submitter does not call the approve function within a
-specific governable number of blocks, anyone can do that and receive the
+specific number of blocks, anyone can do that and receive the
```
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:78]: The period after which everyone becomes eligible to call the approve function will be governable, right? Isn't it missing it from the list?

File: solidity/random-beacon/README.adoc [hunk 5]
```diff
@@ -93,10 +92,10 @@ performed.
 The sortition pool weights operators by their authorized stake amount and allows
 selecting the same operator multiple times. Off-chain DKG protocol executes in
 the same way as for v1 and inactive/disqualified members during the off-chain
-protocol are marked as ineligible for rewards for a certain, governable, period
+protocol are marked as ineligible for rewards for a certain period
```
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:95]: Is this period controlled by `sortitionPoolRewardsBanDuration` variable? If so, could we add add this case to the description of the parameter in the Parameters table?

File: solidity/random-beacon/README.adoc [hunk 6]
```diff
@@ -289,6 +286,144 @@ operators whose nodes never participate in signing because they are
...
+|Yes
+d|`1000e18` +
+_1 000 T_
+
+4+s|Random Beacon
+
+|callbackGasLimit
+|Relay entry callback gas limit
+|Yes
+|
```
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:369]: Earlier in the document we say
> The gas limit for __beaconCallback is initially set to 50k gas 

Can we set default value here in the table to 50k?

File: solidity/random-beacon/README.adoc [hunk 7]
```diff
@@ -289,6 +286,144 @@ operators whose nodes never participate in signing because they are
...
+
+|unauthorizedSigningSlashingAmount
+|Slashing amount when an unauthorized signing has been proved
+|Yes
+|
+
+|sortitionPoolRewardsBanDuration
+|Duration of the sortition pool rewards ban imposed on operators who missed their
+turn for relay entry or DKG result submission
+|Yes
```
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:404]: My understanding is that members of the sortition pool can be punished with temporary exclusion from earning rewards as a result of:
  - inactivity or disqualification during off-chain DKG
  - misbehaviour during the DKG submission
  - failed heartbeat

File: solidity/random-beacon/README.adoc [hunk 8]
```diff
@@ -289,6 +286,144 @@ operators whose nodes never participate in signing because they are
...
+|
+
+|unauthorizedSigningSlashingAmount
+|Slashing amount when an unauthorized signing has been proved
+|Yes
+|
+
+|sortitionPoolRewardsBanDuration
+|Duration of the sortition pool rewards ban imposed on operators who missed their
+turn for relay entry or DKG result submission
```
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:403]: My understanding is that members of the sortition pool can be punished with temporary exclusion from earning rewards as a result of:

* inactivity or disqualification during off-chain DKG
* misbehaviour during the DKG submission
* failed heartbeat

**Review by tomaszslabon — DISMISSED** [review]

File: solidity/random-beacon/README.adoc (hunk 6 above)
Comment (nkuba) [reply solidity/random-beacon/README.adoc:364]: Sure! It's 56k now actually.

File: solidity/random-beacon/README.adoc (hunk 2 above)
Comment (nkuba) [reply solidity/random-beacon/README.adoc:387]: Nice catch! We slash the submitter!

File: solidity/random-beacon/README.adoc (hunk 8 above)
Comment (nkuba) [reply solidity/random-beacon/README.adoc:398]: Yet another nice catch!

File: solidity/random-beacon/README.adoc [hunk 9]
```diff
@@ -284,6 +281,175 @@ operators whose nodes never participate in signing because they are
...
+_1 000 T_
+
+|sortitionPoolUnlockingReward
+|Reward in T for unlocking the sortition pool if DKG timed out
+|Yes
+d|`1000e18` +
+_1 000 T_
+
+|ineligibleOperatorNotifierReward
+|Reward in T for notifying the operator is ineligible
```
[the remark this answers is not in the document]
Comment (michalinacienciala) [reply solidity/random-beacon/README.adoc:396]: BTW, this is out of scope of this PR, but I have some doubts if we should have that reward implemented alongside ETH reimbursement that we do for this operation. I wrote about this param here: https://github.com/keep-network/keep-core/issues/2824#issuecomment-1049979893.

File: solidity/random-beacon/README.adoc [hunk 10]
```diff
@@ -284,6 +281,175 @@ operators whose nodes never participate in signing because they are
...
+|Yes
+|`20 blocks`
+
+4+s|Groups
+
+|relayRequestFee
+|Fee paid by the relay requester
+|Yes
+d|`200e18` +
+`200 T`
```
Comment (michalinacienciala) [inline solidity/random-beacon/README.adoc:343]: Can we move this under `Relay Entry` section?

File: solidity/random-beacon/README.adoc (hunk 9 above)
[the remark this answers is not in the document]
Comment (michalinacienciala) [reply solidity/random-beacon/README.adoc:396]: OK, after discussing it with the team, it looks that this reward will be discarded in the newer implementation of the Random Beacon (approved maintainers sending notification about ineligible operators will be reimbursed for the cost of used gas and there will be no separate reward for sending this tx). I'm ok with leaving this `ineligibleOperatorNotifierReward` param here in this PR and removing it in the PR which will adjust Random Beacon contract and README file to the solutions used in tbtc-v2.

File: solidity/random-beacon/README.adoc [hunk 11]
```diff
@@ -284,6 +281,175 @@ operators whose nodes never participate in signing because they are
...
+d|`5760 blocks` +
+_~24h assuming 15s block time_
+
+|relayEntrySubmissionFailureSlashingAmount
+|Slashing amount for not submitting relay entry
+|Yes
+d|`1000e18` +
+_1 000 T_
+
+4+s|Random Beacon
```
[the remark this answers is not in the document]
Comment (nkuba) [reply solidity/random-beacon/README.adoc:371]: Those are parameters defined and used in the `RandomBeacon` contract. They are not used in the `DKG` library. I would leave them as they are placed now.

File: solidity/random-beacon/README.adoc [hunk 12]
```diff
@@ -284,6 +281,175 @@ operators whose nodes never participate in signing because they are
...
+_90% of groupSize_
+
+|offchainDkgTime
+|Time in blocks after which DKG result is complete and ready to be published by
+clients
+|No
+d|`72 blocks` +
+_5 * (1 + 5) + 2 * (1 + 10) + 20_
+
+|resultChallengePeriodLength
```
[the remark this answers is not in the document]
Comment (nkuba) [reply solidity/random-beacon/README.adoc:320]: The parameter is defined in `DKG` library and it's named `resultChallengePeriodLength`, I would leave it as is here.

File: solidity/random-beacon/contracts/RandomBeacon.sol [hunk 13]
```diff
@@ -105,11 +105,11 @@ contract RandomBeacon is IRandomBeacon, Ownable {
     /// @notice Reward in T for notifying the operator is ineligible.
     uint256 public ineligibleOperatorNotifierReward;
 
-    /// @notice Slashing amount for supporting malicious DKG result. Every
+    /// @notice Slashing amount for submitting a malicious DKG result. Every
     ///         DKG result submitted can be challenged for the time of
     ///         `dkgResultChallengePeriodLength`. If the DKG result submitted
```
Comment (michalinacienciala) [inline solidity/random-beacon/contracts/RandomBeacon.sol:110]: According to your comment https://github.com/keep-network/keep-core/pull/2751#discussion_r816875050 I think we should change it like this:
```suggestion
    ///         `resultChallengePeriodLength`. If the DKG result submitted
```

File: solidity/random-beacon/README.adoc (hunk 11 above)
[the remark this answers is not in the document]
Comment (michalinacienciala) [reply solidity/random-beacon/README.adoc:371]: Ok, makes sense.

File: solidity/random-beacon/README.adoc (hunk 12 above)
[the remark this answers is not in the document]
Comment (michalinacienciala) [reply solidity/random-beacon/README.adoc:320]: Ok, I see. Then I think we should modify some comments that we have in the contracts, I left separate comments for that...

**Review by michalinacienciala — COMMENTED** [review]
Left small comments

File: solidity/random-beacon/README.adoc (hunk 9 above)
[the remark this answers is not in the document]
Comment (michalinacienciala) [reply solidity/random-beacon/README.adoc]: We no longer have the `ineligibleOperatorNotifierReward`  param. Resolving conversation.

File: solidity/ecdsa/contracts/WalletRegistry.sol [hunk 14]
```diff
@@ -50,11 +50,11 @@ contract WalletRegistry is IRandomBeaconConsumer, Ownable {
...
     address public walletOwner;
 
-    /// @notice Slashing amount for supporting malicious DKG result. Every
-    ///         DKG result submitted can be challenged for the time of DKG's
-    ///         `resultChallengePeriodLength` parameter. If the DKG result submitted
-    ///         is challenged and proven to be malicious, each operator who
-    ///         signed the malicious result is slashed for
+    /// @notice Slashing amount for submitting a malicious DKG result. Every
+    ///         DKG result submitted can be challenged for the time of
+    ///         `dkgResultChallengePeriodLength`. If the DKG result submitted
```
[the remark this answers is not in the document]
Comment (michalinacienciala) [reply solidity/ecdsa/contracts/WalletRegistry.sol]: I changed that in [7c7224d](https://github.com/keep-network/keep-core/pull/2751/commits/7c7224d10cab82d7d2ed811c45bbc2b0123e7a17).

File: solidity/random-beacon/contracts/RandomBeacon.sol (hunk 13 above)
Comment (michalinacienciala) [reply solidity/random-beacon/contracts/RandomBeacon.sol]: I changed that in [7c7224d](https://github.com/keep-network/keep-core/pull/2751/commits/7c7224d10cab82d7d2ed811c45bbc2b0123e7a17).

File: solidity/ecdsa/contracts/WalletRegistryGovernance.sol [hunk 15]
```diff
@@ -946,7 +946,7 @@ contract WalletRegistryGovernance is Ownable {
     }
 
     /// @notice Get the time remaining until the DKG result submission
-    ///         eligibility delay can be updated.
+    ///         can be updated.
```
Comment (pdyraga) [inline solidity/ecdsa/contracts/WalletRegistryGovernance.sol:949]:
```suggestion
    ///         timeout can be updated.
```

File: solidity/random-beacon/README.adoc [hunk 16]
```diff
@@ -278,6 +275,191 @@ operators whose nodes never participate in signing because they are
...
+|Yes
+d|`1000e18` +
+_1 000 T_
+
+4+s|Random Beacon
+
+|_owner
+|Address of the RandomBeacon contract owner.
+|Yes
+d|_deployer's address_
```
Comment (pdyraga) [inline solidity/random-beacon/README.adoc:362]: I would remove it, it is confusing. Owner is `RandomBeaconGovernance` contract.

**Review by pdyraga — APPROVED** [review]
I'll do the required changes when working on #2931

**Pull request merged**